### DISCUSSION TITLE
[Feature] Able to disable feature like machine learning.

### DISCUSSION THREAD

**Discussion by troilus** [immich-app/immich#2286] [Q&A] [answered]
### The feature

Am I able to disable some feature like machine learning, because of my old device that the CPU doesn't support AVX or AVX2.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by samip5** [comment, marked as answer]
Yes. https://immich.app/docs/FAQ#why-is-immich-slow-on-low-memory-systems-like-the-raspberry-pi

**Reply by samip5** [reply]
Yes, but it's non-documented yet.

TYPESENSE_ENABLED=false and it will not choke on it.

**Reply by samip5** [reply]
Actually, it was quite easy to find via search: https://github.com/search?q=repo%3Aimmich-app%2Fimmich%20TYPESENSE_ENABLE&type=code

**Reply by troilus** [reply]
Thanks a lot! I found it.

**Reply by jon-f-novabackup** [reply]
> Yes, but it's non-documented yet.
> 
> TYPESENSE_ENABLED=false and it will not choke on it.

Isn't typesense different than machine learning though? I'm trying to understand why would you ever want or need to disable typesense on a Rpi platform, is that really something that should ever need to be disabled, and if it is disabled wouldn't that break the general ability to search even for manually tagged images? I get why you would want to disable machine learning though to set it to false in the .env file and not deploy it as a container on an armv7 Pi.

**Reply by samip5** [reply]
when you currently type anything into the explore pages search field, it actually uses clip embeddings (from machine learning), only with m: it will use description that was how it was explained by @bo0tzz in Discord I think.
